src/App/Fossa/Snippets/Analyze.hs

```diff
@@ -0,0 +1,45 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module App.Fossa.Snippets.Analyze (
+  analyzeWithMillhone,
+) where
+
+import App.Fossa.Config.Snippets (AnalyzeConfig (..), labelForKind, labelForTarget, labelForTransform)
+import App.Fossa.EmbeddedBinary (BinaryPaths, toPath, withMillhoneBinary)
+import App.Types (BaseDir (unBaseDir))
+import Control.Algebra (Has)
+import Control.Effect.Diagnostics (Diagnostics)
+import Control.Effect.Lift (Lift)
+import Data.String.Conversion (toText)
+import Effect.Exec (AllowErr (Never), Command (..), Exec, argFromPath, argsLabeled, execEffectful)
+import Effect.Logger (Logger)
+import Path (Abs, Dir, Path)
+
+analyzeWithMillhone ::
+  ( Has (Lift IO) sig m
+  , Has Exec sig m
+  , Has Diagnostics sig m
+  , Has Logger sig m
+  ) =>
+  AnalyzeConfig ->
+  m ()
+analyzeWithMillhone conf = withMillhoneBinary $ \bin -> execEffectful root $ mkCmd bin root conf
+  where
+    root = unBaseDir $ analyzeScanDir conf
+
+mkCmd :: BinaryPaths -> Path Abs Dir -> AnalyzeConfig -> Command
+mkCmd bin root AnalyzeConfig{..} =
```

Comment:
I think it would be worth adding some tests for `mkCmd` here and for Snippets/Commit.hs too

Reply:
IMO tests for what amounts to argument shuffling don't make a lot of sense (I don't write them for any application, nor for `main` functions, which is effectively what this is).

That aside, we don't actually have a great way to test this today because we'd need to run an actual analysis inside the test to do so. I've filed this ticket to add the ability to test passed arguments: https://fossa.atlassian.net/browse/ANE-1235

Reply:
Going to go ahead and merge this for now but happy to talk more on this and do a follow up PR if needed!

Reply:
Yeah, I'm fine with merging without this. This is on the edge of needing testing and it's not worth doing if it's hard to do.

I think you could do it, though, by exporting `mkCmd` and just testing that. I do that for `themisFlags`, which is pretty much the same thing as `mkCmd`, here: https://github.com/fossas/fossa-cli/blob/master/test/App/Fossa/RunThemisSpec.hs